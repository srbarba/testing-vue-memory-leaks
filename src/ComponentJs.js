export default {
  props: {
    text: {
      type: String,
      default: 'Dummy component',
    },
  },
  render(h) {
    return h('v-card', [this.text, h('div', this.$t('Hello world').toString())])
  },
}
