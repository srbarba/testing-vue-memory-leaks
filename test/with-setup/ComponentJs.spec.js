import { renderWithVuetify } from '../withVuetifyUtils'
import ComponentJs from '../../src/ComponentJs'

describe('ComponentJs', () => {
  it('renders correctly', () => {
    const wrapper = renderWithVuetify(ComponentJs)
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('renders correctly with props', () => {
    const wrapper = renderWithVuetify(ComponentJs, {
      propsData: {
        text: 'My dummy Component',
      },
    })
    expect(wrapper.html()).toMatchSnapshot()
  })
})
