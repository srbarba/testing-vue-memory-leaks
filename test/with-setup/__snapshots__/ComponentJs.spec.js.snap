// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ComponentJs renders correctly 1`] = `
<div class="v-card v-sheet theme--light">Dummy component<div>Hello world</div>
</div>
`;

exports[`ComponentJs renders correctly with props 1`] = `
<div class="v-card v-sheet theme--light">My dummy Component<div>Hello world</div>
</div>
`;
